services/common/src/components/documents/spatial/AddSpatialDocumentsModal.tsx

```diff
@@ -0,0 +1,177 @@
+
+const AddSpatialDocumentsModal: FC<AddSpatialDocumentsModalProps> = ({
+  formName,
+  fieldName,
+  uploadUrl,
+  transformFile,
+}) => {
+  const dispatch = useDispatch();
+  const initialValues = useSelector(getFormValues(formName));
+  const initialDocuments = initialValues[fieldName] ?? [];
+  const modalFormName = `${formName}_${fieldName}`;
+  const currentValues = useSelector(getFormValues(modalFormName));
+  const existingDocuments = currentValues ? currentValues[fieldName] : [];
+  const spatialAcceptedFileTypesMap = {
+    ...OTHER_SPATIAL,
+    ...XML,
+  };
+
+  const [currentStep, setCurrentStep] = useState(0);
+  const [isResetting, setIsResetting] = useState(false);
+
+  const handleFileLoad = (document_name, document_manager_guid, version) => {
+    const fileData = { document_name, document_manager_guid, version };
+    const newUploadedFiles = [
+      ...existingDocuments,
+      transformFile ? transformFile(fileData) : fileData,
+    ];
+    dispatch(change(modalFormName, fieldName, newUploadedFiles));
+    dispatch(touch(modalFormName, fieldName));
+  };
+
+  const handleRemoveFile = (err, fileItem) => {
+    const newUploadedFiles = existingDocuments.filter(
+      (doc) => doc.document_manager_guid !== fileItem.serverId
+    );
+    dispatch(change(modalFormName, fieldName, newUploadedFiles));
+  };
+
+  const stepContent = [
+    {
+      title: "Upload Spatial Data Files",
+      content: (
+        <>
+          <Typography.Title level={3}>Upload Spatial Bundle</Typography.Title>
+          <Typography.Paragraph>
+            You can only submit one KML, KMZ, or Shapefile at a time.
+          </Typography.Paragraph>
+          <Typography.Paragraph>
+            <Alert
+              type="info"
+              showIcon
+              description={
+                <>
+                  Visit <Link to={SPATIAL_DATA_STANDARDS_URL}>GIS Shapefile Standards</Link> to
+                  learn more about shapefile requirements and standards.
+                </>
+              }
+            />
+          </Typography.Paragraph>
+          <Typography.Title level={4}>Spatial Data</Typography.Title>
+          <Typography.Paragraph>
+            Shapefile component files must include the .shp, .shx, .dbf, and .prj files with the
+            same filename prefix.
+          </Typography.Paragraph>
+        </>
+      ),
+    },
+    {
+      title: "Review",
+      content: (
+        <>
+          <Typography.Title level={3}>Review Map</Typography.Title>
+          <Typography.Paragraph>
+            Please confirm that the composed map below correctly represents your selected spatial
+            data files.
+          </Typography.Paragraph>
+          <ViewSpatialDetail spatialDocuments={existingDocuments} />
+        </>
+      ),
+    },
+  ];
+
+  const isFinalStep = currentStep === stepContent.length - 1;
+
+  return (
+    <FormWrapper
+      isModal
+      name={modalFormName}
+      onSubmit={async (values) => {
```

Comment:
Not a huge deal, but I this function is getting a bit large to live inline.  Might be nice as it's own function.

Reply:
Agreed. Extracted a function!